Dockerfile

```diff
@@ -1,16 +1,23 @@
-FROM alpine:3.17 AS base
+FROM golang:1.20 AS build
 
 WORKDIR /app
 
-FROM base AS builder
+# Make sure the dependency downloading can be cached
+COPY go.mod go.sum ./
+RUN go mod download
 
 COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go build -o ./vulnbot
 
-RUN apk update && apk upgrade && apk add go
-RUN go build .
+FROM build as test
+RUN go test -v -race ./...
 
-FROM base AS final
+# Final image uses a barebones image
+FROM gcr.io/distroless/base-debian11 AS release
```

Comment:
[q] If the goal is minimal final image, I've also seen `scratch` used. Thoughts on `distroless` vs `scratch`?

Reply:
Hmmmm yeah I hadn't thought of that. I know that `scratch` is generally what's used for building base images, such as alpine or distroless. I could potentially experiment with that.